spec/rails_helper.rb

```diff
@@ -214,10 +211,20 @@ def seed_base_data_for_tests
     Capybara.server = :puma, { Silent: true }
   end
 
-  config.before(:each) do
-    # Defined shared @ global variables used throughout the test suite.
-    define_global_variables if RSpec.current_example.metadata[:seed_items] != false
+  config.before(:all) do
+    unless self.class.metadata[:skip_seed]
+      seed_base_data_for_tests
+      define_global_variables
+    end
+  end
 
+  config.after(:all) do
+    unless self.class.metadata[:skip_seed]
+      DatabaseCleaner.clean_with(:truncation)
```

Comment:
Did you want to try to do my suggestion where we keep track of the previous suite (if it had skip\_seed on or off)? If it's off for both of them, we don't need to seed data or clean. If it was off previously and on now, we have to truncate beforehand. And if it was on previously and off now, we need to re-seed and don't need to truncate.

Reply:
Yea I'll explore it. I forgot about that part with the metadata thing. I imagine I should be able to set values on the RSpec object from the before part.

Reply:
I have no clue how to do this besides some monkey patching. I can't figure out how to access suite level metadata within  a `before(:all)` hook and the `before(:all)` metadata does not persist between runs. I am open to suggestions.